### PR TITLE
e2e: integrate light clients

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: #5935

Adds light clients to the e2e framwork. 

This is implemented by running the light client programatically. The full node config is mapped to the light client config i.e. persistent peers = providers

### CODE REVIEW

File: light/proxy/proxy.go [hunk 1]
```diff
@@ -21,6 +23,26 @@ type Proxy struct {
 	Listener net.Listener
 }
 
+func NewProxy(
```
Comment (melekes) [inline light/proxy/proxy.go:26]: godoc?

File: light/rpc/client.go [hunk 2]
```diff
@@ -59,6 +60,24 @@ func KeyPathFn(fn KeyPathFunc) Option {
 	}
 }
 
+func DefaultMerkleKeyPathFn() KeyPathFunc {
```
Comment (melekes) [inline light/rpc/client.go:63]: godoc?

File: test/e2e/pkg/testnet.go [hunk 3]
```diff
@@ -327,6 +328,9 @@ func (n Node) Validate(testnet Testnet) error {
 	default:
 		return fmt.Errorf("invalid ABCI protocol setting %q", n.ABCIProtocol)
 	}
+	if n.Mode == ModeLight && n.ABCIProtocol != ProtocolBuiltin {
+		return errors.New("light client must used built in protocol")
```
Comment (melekes) [inline test/e2e/pkg/testnet.go:332]:
```suggestion
		return errors.New("light client must use built in protocol")
```

File: light/client.go [hunk 4]
```diff
@@ -861,7 +861,7 @@ func (c *Client) cleanupAfter(height int64) error {
 }
 
 func (c *Client) updateTrustedLightBlock(l *types.LightBlock) error {
-	c.logger.Debug("Updating trusted light block", "light block", l)
+	c.logger.Debug("updating trusted light block", "light block", l)
```
Comment (melekes) [inline light/client.go:864]: this won't get logged because logger does not allow whitespaces in key names ("light block")

**Comment by cmwaters** [comment]
I have a hunch that the addition of  the light clients has increased the burden on the network. This hunch comes about as I see more frequent errors arise due to nodes failing to reach a certain height at a certain time i.e.

```
timed out waiting for full02 to reach height 1025 
```

However, I don't think this is purely concerned with light clients. I am also observing this sometimes with the current `master` branch. I'm not sure whether to loosen the current timeouts or to reduce the involvement of the light clients in the network i.e. to not send txs through light clients

**Review by melekes — APPROVED** [review]
👍

**Pull request merged**